xds/src/main/java/com/linecorp/centraldogma/xds/k8s/v1/XdsKubernetesEndpointFetchingService.java

```diff
@@ -241,24 +236,41 @@ private static class KubernetesEndpointsUpdater {
                     if (endpoints.isEmpty()) {
                         return;
                     }
-                    executorService.execute(this::maybeSchedule);
+                    executorService.execute(this::maybePushK8sEndpoints);
                 }, true);
             }));
         }
 
-        void maybeSchedule() {
-            if (closing || scheduledFuture != null) {
+        private void maybePushK8sEndpoints() {
+            if (closing) {
                 return;
             }
-            // Commit after 3 seconds so that it pushes with all the fetched endpoints in the duration
-            // instead of pushing one by one.
-            scheduledFuture = executorService.schedule(() -> {
-                scheduledFuture = null;
-                if (closing) {
-                    return;
-                }
+            if (allEndpointGroupsReady()) {
+                // Push only when all the endpoint groups are ready.
                 pushK8sEndpoints();
-            }, 3, TimeUnit.SECONDS);
+            } else {
+                // Schedule it again to prevent the situation where the last endpoint group
+                // is completed exceptionally.
+                executorService.schedule(this::maybePushK8sEndpoints, 3, TimeUnit.SECONDS);
```

Comment:
Optional) 3 seconds seems too long to wait for `KubernetesEndpointGroup` to be updated. 

How about using the callback style instead?
```java
@Nullable
private CompletableFuture<?> allEndpointGroupsReady() {
    if (!kubernetesEndpointGroupFuture.isDone()) {
        return null;
    }

    ....
    // If KubernetesEndpointGroups are not ready. 
    CompletableFuture.allOf(...);
}
```

Reply:
> CompletableFuture.allOf(...);

The future is completed when one of the futures is completed exceptionally. I need to wait for all. 😉

Reply:
Changed to use `successfulAsList`. PTAL. 😉